src/video/SDL_egl.c

```diff
@@ -79,6 +79,15 @@
 #define DEFAULT_OGL_ES_PVR "libGLES_CM.so"
 #define DEFAULT_OGL_ES "libGLESv1_CM.so"
 
+#elif SDL_VIDEO_DRIVER_KMSDRM
+/* Non-desktop Linux must NOT depend on X11/GLX for FULL OpenGL,
+   it must use GLVND instead so it's libOpenGL.so instead of libGL.so. */
+#define DEFAULT_OGL "libOpenGL.so"
+#define DEFAULT_EGL "libEGL.so"
+#define DEFAULT_OGL_ES2 "libGLESv2.so"
+#define DEFAULT_OGL_ES_PVR "libGLES_CM.so"
+#define DEFAULT_OGL_ES "libGLESv1_CM.so"
```

Comment:
I suppose this is going to change the behavior of SDL when it's compiled with `--enable-video-kmsdrm` even if the active video backend is X11 or Wayland (such as what may be the case for applications that want to support running from the console and within a desktop environment). It looks it already does change the default libraries when built with `--enable-video-rpi` even if it's not running on a Pi.

Maybe the code here should examine the actual backend at runtime rather than hardcoding it based on build-time variables?

Reply:
@cgutman But that's inside `#elif SDL_VIDEO_DRIVER_KMSDRM`, how's that affecting the `--enable-raspberry-pi` builds?

What `#elif SDL_VIDEO_DRIVER_KMSDRM` is precissely that: it examines the actual backend and activates the usage of GLVND dlls only if it's KMSDRM.

Reply:
But those are compile-time checks. They don't look at the actual video backend selected at runtime.

Consider 2 builds of SDL run with different `configure` options:
1. `./configure` (default - KMSDRM off)
2. `./configure  --enable-video-kmsdrm`

Now if I run this build _on X11_, I will end up using `libGL.so.1` on build 1 and `libOpenGL.so` on build 2, even though I'm not actually using KMSDRM. I only have it enabled at compile-time.

This isn't a purely theoretical concern either. The Debian Bullseye package is built with `--enable-video-kmsdrm` https://packages.debian.org/bullseye/libsdl2-2.0-0 so this would change the default OGL library for them on X11/Wayland too if they pulled in this change.

Reply:
Oh, I understand now. GLVND (libOpenGL.so) seems to work for both X11 and KMSDRM here (in a  `--enable-video-kmsdrm` build) so I thought libOpenGL.so was appropiated for everything if KMSDRM support is built-in. But this is incoherent: using libGL.so when kmsdrm support is not built in and libOpenGL.so when it is on, seems to be just wrong.

Well, I need some help with the solution then.
What would be a good place to put the logic to chose between libGL.so or libOpenGL.so depending on the backend in use? I imagine this function, am I right?

https://github.com/libsdl-org/SDL/blob/9d46bd25c7db0fcfaf70ac7d33b4058648e057ab/src/video/SDL_egl.c#L289

Also, any idea on how can I find out the backend in use from there?